### DISCUSSION TITLE
How do I get a Pinecone Namespace?

### DISCUSSION THREAD

**Discussion by Arche151** [mayooear/ai-pdf-chatbot-langchain#41] [General]
Hey all, 

in the README, there are instructions to update the PINECONE_INDEX_NAME and PINECONE_NAME_SPACE in the config folder of the project. I have an index name in my Pinecone dashboard, but I don't have a namespace. My Pinecone dashboard says:

Empty Index
Upsert Vectors to view index info

Could you please guide me on how to get or set the "NAMESPACE" in Pinecone, so that I can proceed with the project setup? Any help would be greatly appreciated. Thank you!

**Comment by Aditya1404Sal** [comment]
it's an free naming scheme , you can assign the value of 
const PINECONE_NAME_SPACE as anything , it doesn't affect the dev model
for example 
'demo' or the default 'pdf-test'

**Comment by mayooear** [comment]
> Hey all,
> 
> in the README, there are instructions to update the PINECONE_INDEX_NAME and PINECONE_NAME_SPACE in the config folder of the project. I have an index name in my Pinecone dashboard, but I don't have a namespace. My Pinecone dashboard says:
> 
> Empty Index Upsert Vectors to view index info
> 
> Could you please guide me on how to get or set the "NAMESPACE" in Pinecone, so that I can proceed with the project setup? Any help would be greatly appreciated. Thank you!

When you ingest the namespace is created based on the value you gave to the namespace variable in `config/pinecone.ts`.

Once it's stored in Pinecone, that same namespace is used to query it.
